### PR TITLE
fix: yaml indentation in statefulset for `extraEnvs`, `envFrom` and `extraVolumeMounts`

### PR DESCRIPTION
**Pull request by sastorsl**
Signed-off-by: Stein Arne Storslett <sastorsl@users.noreply.github.com>

### Description
Fix toYaml error when adding `extraEnvs`, `envFrom` and `extraVolumeMounts` to `values.yaml`.

Passes `helm lint -f values.yaml ./` and install after adding `extraEnvs` after change.

### Issues Resolved
This PR resolves issue #52, #64 and #65
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by mprimeaux** [comment]
@sastorsl Hi there. Would you mind reviewing the diff in issue #64? I believe there may be one more place to change the indent from 10 to 8?

**Comment by sastorsl** [comment]
> @sastorsl Hi there. Would you mind reviewing the diff in issue #64? I believe there may be one more place to change the indent from 10 to 8? The diff is at the top in a collapsable div. Thanks in advance.

I believe you are right - and that it applies to `envFrom`, `initResources` - and perhaps others as well.

Could be fixed in the same PR, but would that be OK with regards to you contribution policy?

**Comment by sastorsl** [comment]
@mprimeaux I've added a second commit, including `extraVolumeMounts` (#65).
Would be great if you took a look at it, thanks!

Vent through some of the other values as well, but `envFrom`, `extraEnvs` and `extraVolumeMounts` where the ones I found affected at this time.

<details>
  <summary>values.yaml diff</summary>
  
```diff
diff --git a/charts/opensearch/values.yaml b/charts/opensearch/values.yaml
index b62bb8e..d455176 100755
--- a/charts/opensearch/values.yaml
+++ b/charts/opensearch/values.yaml
@@ -83,16 +83,20 @@ config:
 # Extra environment variables to append to this nodeGroup
 # This will be appended to the current 'env:' key. You can use any of the kubernetes env
 # syntax here
-extraEnvs: []
+extraEnvs:
+- name: test
+  value: me
 #  - name: MY_ENVIRONMENT_VAR
 #    value: the_value_goes_here
 
 # Allows you to load environment variables from kubernextes secret or config map
-envFrom: []
+envFrom:
 # - secretRef:
 #     name: env-secret
 # - configMapRef:
 #     name: config-map
+- secretRef:
+    name: env-secret
 
 # A list of secrets and their paths to mount inside the pod
 # This is useful for mounting certificates for security and for mounting
@@ -123,21 +127,21 @@ resources:
     cpu: "1000m"
     memory: "100Mi"
 
-initResources: {}
-  # limits:
-  #   cpu: "25m"
+initResources:
+  limits:
+    cpu: "25m"
   #   # memory: "128Mi"
   # requests:
   #   cpu: "25m"
   #   memory: "128Mi"
 
-sidecarResources: {}
-  # limits:
-  #   cpu: "25m"
-  #   # memory: "128Mi"
-  # requests:
-  #   cpu: "25m"
-  #   memory: "128Mi"
+sidecarResources:
+ limits:
+   cpu: "25m"
+   # memory: "128Mi"
+ requests:
+   cpu: "25m"
+   memory: "128Mi"
 
 networkHost: "0.0.0.0"
 
@@ -183,14 +187,14 @@ persistence:
   size: 8Gi
   annotations: {}
 
-extraVolumes: []
-  # - name: extras
-  #   emptyDir: {}
+extraVolumes:
+- name: extras
+  emptyDir: {}
 
-extraVolumeMounts: []
-  # - name: extras
-  #   mountPath: /usr/share/extras
-  #   readOnly: true
+extraVolumeMounts:
+- name: extras
+  mountPath: /usr/share/extras
+  readOnly: true
 
 extraContainers: []
   # - name: do-something
@@ -344,7 +348,8 @@ lifecycle: {}
   #         while [[ "$(curl -s -o /dev/null -w '%{http_code}\n' $ES_URL)" != "200" ]]; do sleep 1; done
   #         curl -XPUT "$ES_URL/_template/$TEMPLATE_NAME" -H 'Content-Type: application/json' -d'{"index_patterns":['\""$INDEX_PATTERN"\"'],"settings":{"number_of_shards":'$SHARD_COUNT',"number_of_replicas":'$REPLICA_COUNT'}}'
 
-keystore: []
+keystore:
+- secretName: "foobar"
 
 networkPolicy:
   ## Enable creation of NetworkPolicy resources. Only Ingress traffic is filtered for now.

```
</details>

**Comment by mprimeaux** [comment]
@sastorsl Yes, I can confirm that `extraVolumeMounts` should also be set to 8. Here is the error with indentation set to 10:

```shell
Error: INSTALLATION FAILED: YAML parse error on logging/charts/opensearch/templates/statefulset.yaml: error converting YAML to JSON: yaml: line 138: did not find expected key
make: *** [install] Error 1
```

Thanks again for the review and fix.

**Comment by sastorsl** [comment]
How do I go about to get this reviewed and merged?

**Comment by peterzhuamazon** [comment]
Added @DandyDeveloper @TheAlgo to take a look, @smlx @mprimeaux would like your approval.
We are currently a bit busy with next upcoming release here, so recently the repo activities here are a bit slow :)
I apologize for this. 

Thanks.

**Review by DandyDeveloper — APPROVED** [review]
@sastorsl Since we have a CI in place now, we'll need a chart bump for now. 

I think changelog can wait until we've ironed out things.

**Comment by mprimeaux** [comment]
@peterzhuamazon @DandyDeveloper I coordinated with @sastorsl on this and have tested in our environment. LGTM

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon @DandyDeveloper I coordinated with @sastorsl on this in terms of testing in our environment. LGTM aside from the increment of the chart version to `1.0.4`.

Havent check yet but is version bump part of our developer guide?
We probably need to add it in case people forget.

**Comment by mprimeaux** [comment]
> Havent check yet but is version bump part of our developer guide?
We probably need to add it in case people forget.

@peterzhuamazon This is a welcome addition to the version guide. Good catch.

**Comment by TheAlgo** [comment]
> > Havent check yet but is version bump part of our developer guide?
> > We probably need to add it in case people forget.
> 
> @peterzhuamazon This is a welcome addition to the version guide. Good catch. If you and the maintainers are supportive then I don't mind issuing a PR for the change.

Yes , let's add proper guidelines for it. It will be great

**Review by TheAlgo — APPROVED** [review]

**Comment by TheAlgo** [comment]
@sastorsl Lint checks are failing can you cross check once?

**Comment by mprimeaux** [comment]
@sastorsl The related chart version needs to be bumped to `1.0.4`.

**Comment by peterzhuamazon** [comment]
> @sastorsl The related chart version needs to be bumped to `1.0.4` [here](https://github.com/opensearch-project/helm-charts/blob/6eb57792ebfe49abb1297923ecb200f5f9bfced1/charts/opensearch/Chart.yaml#L18).

@mprimeaux I wonder if we need to get a github action auto bumper after we click merge.
It will be much easier and will not have edge cases where multiple PR change to the same 1.x.x version and conflict again if any of them get merged.

**Comment by mprimeaux** [comment]
@peterzhuamazon I will research and come back with a few options. The challenge is the 'auto-bumper' would need to know which of the version components to increment as per Semantic Versioning 2.0 rules. For example, breaking change requires a major version bump. Non-breaking features requires a minor version bump. Defect fixes require a patch version number bump.

If you'd like, for now, I can commit the version bump since it seems @sastorsl is busy with other 'stuff'.  Please let me know.

**Comment by peterzhuamazon** [comment]
> @peterzhuamazon I will research and come back with a few options. The challenge is the 'auto-bumper' would need to know which of the version components to increment as per Semantic Versioning 2.0 rules. For example, breaking change requires a major version bump. Non-breaking features requires a minor version bump. Defect fixes require a patch version number bump.
> 
> If you'd like, for now, I can commit the version bump since it seems @sastorsl is busy with other 'stuff'. Please let me know as we are blocked waiting for this PR to merge.

If @sastorsl has you added to his fork, then cool I am ok with you commit.
Seems like there are other community members waiting for this fix.
Thanks.

**Comment by mprimeaux** [comment]
@peterzhuamazon I don't personally know @sastorsl and completely forgot that I'd need commit access to his fork. One option, not ideal, is I can fork and make the changes and submit a PR since this is time-critical for many folks. Thoughts?

**Comment by mprimeaux** [comment]
@peterzhuamazon Another option is I could open a PR just to bump the chart version to `1.0.4` and then you can merge this one followed by the new PR to bump the version. Just reasoning through options.

**Comment by TheAlgo** [comment]
>@peterzhuamazon I don't personally know @sastorsl and completely forgot that I'd need commit access to his fork. One option, not ideal, is I can fork and make the changes and submit a PR since this is time-critical for many folks. I'm open to any and all suggestions.

@mprimeaux In the interest of time I feel the first option is better instead of merging this PR in case @sastorsl is busy.

**Comment by mprimeaux** [comment]
@TheAlgo I will make the changes and submit now. Stand by.

**Comment by mprimeaux** [comment]
@TheAlgo PR #80 has been submitted to replace this PR.  Please note that I had to increment the `opensearch-dashboards` chart since PR #75 has not been merged.

**Comment by sastorsl** [comment]
It seems I missed a deadline on this one? I'm away for the weekend, and away from my PC until Monday.

**Comment by DandyDeveloper** [comment]
@sastorsl I don't think you've missed anything in that respect, but there's another PR that'll effectively achieve the same thing here submitted by @mprimeaux

**Comment by TheAlgo** [comment]
Closing this in favour of #80 . Apologies @sastorsl but there was nothing wrong , just as this PR was a pain point so we had to address this earlier. Lint checks were not passing for this PR. Thanks a lot @sastorsl 😁

**Pull request closed without merging**
